File: python/Leetcode104.py
from typing import Optional
from Utils import TreeNode
class Leetcode104:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def traverse(root, level):
            if root is None:
                return level
            return max(traverse(root.left, level + 1), traverse(root.right, level + 1))
        
        return traverse(root, 0)

File: java/src/Leetcode104.java
import utils.TreeNode;

public class Leetcode104 {
    public int maxDepth(TreeNode root) {
        return traverse(root, 0);
    }
    private int traverse(TreeNode root, int level){
        if(root == null)
            return level;
        return Math.max(traverse(root.left, level + 1), traverse(root.right, level + 1));
    }
}
